### PR TITLE
CS: rename all classes to comply with the rules

### PR DESCRIPTION
**Pull request by jrfnl**
This PR replaces the previous PR #55.

In contrast to PR #55, this PR now includes namespacing of the classes.

## Description

Compliance with the YoastCS/WPCS class name rules, including the prefixing rules.

Includes:
* Adjusting existing references to the old class names within the plugin.
* Putting a fallback class in place for each class with the old class name extending the new class name to not break BC.
    This allows for throwing deprecation notices if there would happen to be another plugin which extends the (old) class.

As discussed with @herregroen on Slack, all the "fallback classes" have been put into one (new) file `deprecated/deprecated-classes.php`.

### Notes
1. I've removed the `Hacks` from the class name for `Admin`, `Email_Links` and `Forms`.
2. I've added `Comment` to the class name for `Clean_Emails` to comply with the plugin prefixing rules.

### For comparison:

Folder | Old Class Name | New Class Name
-- | -- | --
admin | YoastComment**Hacks**Admin | Yoast\WP\Comment\Admin\Admin
admin | YoastCommentParent | Yoast\WP\Comment\Admin\Comment_Parent
inc | YoastCleanEmails | Yoast\WP\Comment\Inc\Clean_Emails
inc | YoastComment**Hacks**EmailLinks | Yoast\WP\Comment\Inc\Email_Links
inc | YoastCommentForm**Hacks** | Yoast\WP\Comment\Inc\Forms
inc | YoastCommentHacks | Yoast\WP\Comment\Inc\Hacks
inc | YoastCommentLength | Yoast\WP\Comment\Inc\Length
inc | YoastCommentNotifications | Yoast\WP\Comment\Inc\Notifications

## Code Review

This PR will be easiest to review by inspecting each commit individually.

## Testing

Testing this change is highly recommended!

To test it:
* Make sure you have an install with this plugin activated and `WP_DEBUG` = `true`. 
* Switch to this branch and run `composer dump-autoload`.
* Open any page on the website. Most potential problems should be visible straight away with a fatal error for a missing class.
* Check on a blog page with a comment form.
* Check the Comment Hacks admin page.

If all looks as expected and there have been no error notices or white screens, we're good.

### CODE REVIEW

**Comment by jrfnl** [comment]
Note: the build failure is unrelated to this PR, but has to do with the `grunt` vs `phpcs` issue.

File: inc/class-comment-form-hacks.php [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
 
 /**
- * Class YoastCommentFormHacks.
+ * Add comment note.
  *
  * @since 1.3
+ * @since 1.6.0 Class renamed from `YoastCommentFormHacks` to `Yoast\WP\Comment\Inc\Forms`.
  */
-class YoastCommentFormHacks {
+class Yoast_Comment_Forms {
```
Comment (johannadevos) [inline inc/class-comment-form-hacks.php]: This should be just `Forms`, right?

Comment (jrfnl) [reply inc/class-comment-form-hacks.php]: @johannadevos Cricky! Not sure how that one slipped past me. Thank you for catching this. Fixed now.

**Comment by karlijnbok** [comment]
CR + Acceptance: 👍

**Pull request merged**